app/code/Magento/Catalog/Test/Unit/Model/Product/UrlTest.php

```diff
@@ -212,6 +230,18 @@ public function getUrlDataProvider()
                 ['_direct' => '/product/url/path', '_query' => []],
                 null,
                 null,
+                false,
+            ], [
```

Comment:
Thanks for adding a test case to cover the category being set as well as updating the other cases to work with the false flag.